language/move-lang/src/to_bytecode/translate.rs

```diff
@@ -739,12 +727,24 @@ fn abilities(set: &AbilitySet) -> BTreeSet<IR::Ability> {
     set.iter().map(ability).collect()
 }
 
-fn type_parameters(tps: Vec<TParam>) -> Vec<(IR::TypeVar, BTreeSet<IR::Ability>)> {
+fn fun_type_parameters(tps: Vec<TParam>) -> Vec<(IR::TypeVar, BTreeSet<IR::Ability>)> {
     tps.into_iter()
         .map(|tp| (type_var(tp.user_specified_name), abilities(&tp.abilities)))
         .collect()
 }
 
+fn struct_type_parameters(tps: Vec<TParam>) -> Vec<IR::StructTypeParameter> {
+    tps.into_iter()
+        .map(|tp| {
+            (
+                type_var(tp.user_specified_name),
+                abilities(&tp.abilities),
```

Comment:
nit, I think this will be easier to read with some let bindings. So `let name = type_var(tp.user_specified_name); let constraints = abilities(&tp.abilities); (name, constraints, /* is_phantom */ false)`